services/provider/server/server.go

```diff
@@ -1042,3 +1043,130 @@ func (s *OCSProviderServer) PeerStorageCluster(ctx context.Context, req *pb.Peer
 
 	return &pb.PeerStorageClusterResponse{}, nil
 }
+
+func (s *OCSProviderServer) GetStorageClientsInfo(ctx context.Context, req *pb.StorageClientsInfoRequest) (*pb.StorageClientsInfoResponse, error) {
+	response := &pb.StorageClientsInfoResponse{ClientsInfo: []*pb.ClientInfo{}, Error: []*pb.StorageClientInfoError{}}
+
+	for i := range req.ClientIDs {
+		consumer, err := s.consumerManager.Get(ctx, req.ClientIDs[i])
+		if err != nil {
+			if !kerrors.IsNotFound(err) {
+				klog.Errorf("failed to get consumer with id %v: %v", req.ClientIDs[i], err)
+				response.Error = append(response.Error,
+					&pb.StorageClientInfoError{
+						ClientID: req.ClientIDs[i],
+						Code:     pb.ErrorCode_Internal,
+						Message:  err.Error(),
+					},
+				)
+			}
+			continue
+		}
+		owner := util.FindOwnerRefByKind(consumer, "StorageCluster")
+
+		if owner == nil {
+			continue
+		}
+
+		if owner.UID != types.UID(req.StorageClusterUID) {
+			klog.Errorf("storageCluster specified on the req does not own the Client")
+			response.Error = append(response.Error,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_Internal,
+					Message:  "storageCluster specified on the req does not own the Client",
+				},
+			)
+			continue
+		}
+
+		rnsList := &rookCephv1.CephBlockPoolRadosNamespaceList{}
+		err = s.client.List(
+			ctx,
+			rnsList,
+			client.InNamespace(s.namespace),
+			client.MatchingLabels{controllers.StorageConsumerNameLabel: consumer.Name},
+			client.Limit(2),
+		)
+		if err != nil {
+			klog.Error(err)
+			response.Error = append(response.Error,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_Internal,
+					Message:  "failed to get radosnamespaces",
+				},
+			)
+			continue
+		}
+		if len(rnsList.Items) > 1 {
+			response.Error = append(response.Error,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_Internal,
+					Message:  "invalid number of radosnamespace found for the Client",
+				},
+			)
+		} else if len(rnsList.Items) == 1 {
+			response.ClientsInfo = append(response.ClientsInfo, &pb.ClientInfo{ClientID: req.ClientIDs[i], RadosNamespace: rnsList.Items[0].Name})
+		} else {
+			response.ClientsInfo = append(response.ClientsInfo, &pb.ClientInfo{ClientID: req.ClientIDs[i]})
+		}
```

Comment:
1. There is no need to have 2 append statements
2. You should treat the "good" case as a for loop scoped code

```suggestion
		} 

		clientInfo := pb.ClientInfo{ClientID: req.ClientIDs[i]
		if len(len(rnsList.Items) != 0 {
			clientInfo.RadosNamesapce = rnsList[0].Name
		}
		response.ClientsInfo = append(response.ClientsInfo, &clientInfo)
	
```